posts/20230126/index.qmd

```diff
@@ -0,0 +1,45 @@
+* L1.3 Feature selection on training and test set: This is the second most common type of data leakage found in Table 1. 
+* L1.4 Duplicates in dataset: This seems more like an error than a type of data leakage.
+* L2 Model features that aren't legitimate: We spent the most time discussing this category. We found it hard to think about proxy variables because they require a lot of domain knowledge. Instead, we like the definition of a feature the model would not have access to when making new predictions. 
+* L3.1 Temporal leakage: This is when future data is included in model training.
+* L3.2 Non-independence between training and test samples: This can be caused by resampling the same patient or geographic location. This type of leakage can also be due to the interpretation of the results. This occurs when there is a mismatch between the test distribution and the scientific claim. 
+* L3.3 Sampling bias in test distribution: This is choosing a non-representative subset for analysis.
+
+In general, categories L1.1-4 are clear cut and should be identifiable and avoidable. L2 is difficult because it depends on domain knowledge. Due to the nature of collaboration the researcher with this expert domain knowledge may not be the individual performing the ML analysis. Similarly, L3.1-3.3 are also difficult to detect and avoid if the researcher is using previously collected data and not designing an experiment and collecting data from scratch. In industry, data is often given to ML practitioners and there is no option to immediately collect more data or change the collection methodology. If this is the case, data leakage could be undetectable in many industry applications as ML practitioners are working with the best data at hand. In addition, there are limited strategies for identifying and mitigating these three types of data leakage. 
+
+## Model info sheets to mitigate data leakage
+We appreciated that the authors suggested a mitigation strategy. We found the questions in the model info sheets to be a useful sanity check for any ML practitioner. However, the level of detail requested from the info sheets would likely only be filled out if peer reviewed journals required it.  Even if journals require model info sheets as part of the peer review process, we agree with the authors that there are still limitations. Most notably, the claims in the model info sheets can't be verified without the code, computing environment, and data. There are some scientific research fields where authors might be unwilling or unable to disclose all information due to privacy concerns. In addition, there is nothing to prevent authors from lying or not having the ML expertise to fill out the model info sheet correctly. Since ML has become more accessible to non-ML experts, subtle cases might not easily be noticed by the author or reviewer. 
```

Comment:
Yeah, as we discussed, I wish they address more practical aspects of improvement as you mentioned here: "There are some scientific research fields where authors might be unwilling or unable to disclose all information due to privacy concerns."